src/cider_nrepl/main.clj

```diff
@@ -28,17 +28,36 @@
   [middleware-var-strs]
   (into [] mw-xf middleware-var-strs))
 
+(defn- build-handler
```

Comment:
I think that build handler should just take a list of middlewares and be invoked only if a handler wasn't passed to `start-nrepl`.

Reply:
I see, so that implies that either a handler or middleware is expected, but not both. I can make that change.

Reply:
Yeah, exactly. Normally you'd just use a predefined handler, but in some cases you might opt to build the handler yourself by supplying some list of middlewares instead.

Reply:
(e.g. because some middleware is problematic in your setup or something like this)